spec/abci/abci++_app_requirements.md

```diff
@@ -837,6 +846,12 @@ First, some simple start conditions:
 
```

Comment:
Also commenting something that is not part of this PR, ehe.

> Note also CometBFT never calls Commit on an ABCI app twice for the same height.

Comet can call `Commit` for height H, then we have a crash, a recover, and since the application has returned to have state at height H, Comet will call `Commit` for height H "again". So, in fact, if `Commit` has returned for height H, then Comet will never call it again for height H.

Reply:
I guess a more accurate descrption here would be that `Commet` never completes the execution twice? As in, if it crashes before the return of `Commit` (and state persistance), it is technically possible to call it again?

Reply:
Yes. But I didn't want to make this description (on the sentence) too long.

Reply:
Actually, I am not sure anymore, after reading the comment block here: https://github.com/cometbft/cometbft/blob/main/consensus/state.go#L1672

Reply:
Maybe the problem is referring to too things that are too general as synchronization points. I mean, for example, we shouldn't refer to "execute" commit or "return from" commit, but to the atomic operation of updating the height, hash, and the app state inside commit.

Reply:
That's a good point. because in my mind, a successful commit means the state is update as well as all other relevant fields. Maybe we should first specify what exactly is a successful execution of commit (on our side at least).

Reply:
Yes, I think that would work.